src/pages/schedule-interview/components/MobileCandidateSelectInterviewTime/hooks/useTimeSlots.ts

```diff
@@ -0,0 +1,25 @@
+import { useState } from "react"
+
+import { InterviewTimeSlot } from "../type"
+
+const formatTimeSlots = (interviewTimes: InterviewTimeSlot[], currentDate: string): string[] =>
+  interviewTimes
+    .filter((slot) => slot.start.startsWith(currentDate))
+    .map(
+      (slot) =>
+        `${new Date(slot.start).toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" })} - ${new Date(slot.end).toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" })}`
+    )
+export const useTimeSlots = (interviewTimes: InterviewTimeSlot[], initialCurrentDate: string) => {
+  const [timeSlots, setTimeSlots] = useState<string[]>(formatTimeSlots(interviewTimes, initialCurrentDate))
+
+  const updateTimeSlots = (interviewTimes: InterviewTimeSlot[], currentDate: string) => {
+    const slots = formatTimeSlots(interviewTimes, currentDate)
+
+    setTimeSlots(slots)
+  }
+
+  return {
+    updateTimeSlots,
+    timeSlots,
+  }
+}
```

Comment:
I am not sure why we need to store `timeSlots` as `state` here still. We only use the hook in `MobileInterviewTimeSlotBooking`, and we only used this hook to transform the interviewTimes to timeslots right?

```suggestion
export const useTimeSlots = () => {
  const formatTimeSlots = (interviewTimes: InterviewTimeSlot[], currentDate: string): string[] =>
  interviewTimes
    .filter((slot) => slot.start.startsWith(currentDate))
    .map(
      (slot) =>
        `${new Date(slot.start).toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" })} - ${new Date(slot.end).toLocaleTimeString([], { hour: "2-digit", minute: "2-digit" })}`
    )

  return { formatTimeSlots }
}
```

And in `MobileInterviewTimeSlotBooking`, we only need to 
```
  const { formatTimeSlots }  = useTimeSlots()
  const timeSlots = formatTimeSlots(interviewTimes, currentDate)

```

In this way, the hook actually plays the role of a collection of util functions

Reply:
Thanks @martin-luo . I understand what you mean